packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -115,13 +116,13 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ISemver {
 
     /// @notice Reverts when paused.
     modifier whenNotPaused() {
-        require(!paused(), "OptimismPortal: paused");
+        if (paused()) revert CallPaused();
         _;
     }
 
     /// @notice Semantic version.
-    /// @custom:semver 3.5.0
-    string public constant version = "3.5.0";
+    /// @custom:semver 3.6.0
+    string public constant version = "3.6.0";
```

Comment:
Question from @clabby in https://github.com/ethereum-optimism/optimism/pull/10117#pullrequestreview-1995557037:

> This should probably be a major semver bump due to the ABI-breaking changes with removing the legacy getters, but we do have a major bump queued up in the OptimismPortal2. Do we see this being deployed soon, and are we okay with violating proper semver here?

We have some (outdated) semver information in the [style guide](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/STYLE_GUIDE.md#versioning). But it feels weird to bump this to a 4.0.0, given that the latest governance approved release (from MCP L1) is 2.5.0. 

In general, our semver scheme here feels non-standard and confusing for us and users. If the idea is to convey breaking changes to users, these version bumps should correspond to governance released tags. Otherwise it feels like we are releasing a safe version 3.6.0 here, but we are not. This is analogous to an npm package releasing new vX.Y.Z versions during development and separately publishing a list of which ones you should actually use.

I'd expect the next governance-approved portal version here to be 3.0.0 no mater how many times we change the portal. We can use e.g. 3.0.0-beta1 as the style for these WIP versions. This would also make it very explicit when a contract is safe to be used in production vs. when it's not—if it's a normal semver, it's safe. If it has a `-rcX` or `-betaX` it should not be used in prod.

All that to say: IMO we should not bump to 4.0.0 here, and we should reconsider how we handle these semvers

Reply:
I agree on redoing the way that we do semver